internal/federationdomain/endpointsmanager/manager.go

```diff
@@ -163,38 +176,55 @@ func (m *Manager) SetFederationDomains(federationDomains ...*federationdomainpro
 			oauthHelperWithKubeStorage,
 			timeoutsConfiguration.OverrideDefaultAccessTokenLifespan,
 			timeoutsConfiguration.OverrideDefaultIDTokenLifespan,
+			m.auditLogger,
 		)
 
 		m.providerHandlers[(issuerHostWithPath + oidc.PinnipedLoginPath)] = login.NewHandler(
 			upstreamStateEncoder,
 			csrfCookieEncoder,
 			login.NewGetHandler(incomingFederationDomain.IssuerPath()+oidc.PinnipedLoginPath),
-			login.NewPostHandler(issuerURL, idpLister, oauthHelperWithKubeStorage),
+			login.NewPostHandler(issuerURL, idpLister, oauthHelperWithKubeStorage, m.auditLogger),
+			m.auditLogger,
 		)
 
 		plog.Debug("oidc provider manager added or updated issuer", "issuer", issuerURL)
 	}
 }
 
-// ServeHTTP implements the http.Handler interface.
-func (m *Manager) ServeHTTP(resp http.ResponseWriter, req *http.Request) {
-	requestHandler := m.findHandler(req)
-
-	// Using Info level so the user can safely configure a production Supervisor to show this message if they choose.
-	plog.Info("received incoming request",
-		"proto", req.Proto,
-		"method", req.Method,
-		"host", req.Host,
-		"requestSNIServerName", requestutil.SNIServerName(req),
-		"path", req.URL.Path,
-		"remoteAddr", req.RemoteAddr,
-		"foundFederationDomainRequestHandler", requestHandler != nil,
-	)
+func (m *Manager) buildHandlerChain(nextHandler http.Handler) {
+	// build the basic handler for FederationDomain endpoints
+	handler := m.buildManagerHandler(nextHandler)
+	// log all requests, including audit ID
+	handler = requestlogger.WithHTTPRequestAuditLogging(handler, m.auditLogger)
+	// add random audit ID to request context and response headers
+	handler = requestlogger.WithAuditID(handler, func() string {
+		return uuid.New().String()
+	})
+	m.handlerChain = handler
+}
 
-	if requestHandler == nil {
-		requestHandler = m.nextHandler // couldn't find an issuer to handle the request
-	}
-	requestHandler.ServeHTTP(resp, req)
+func (m *Manager) buildManagerHandler(nextHandler http.Handler) http.Handler {
+	return http.HandlerFunc(func(resp http.ResponseWriter, req *http.Request) {
+		requestHandler := m.findHandler(req)
+
+		// TODO: Should this old log message change in light of the new audit logs? Or do we not want to force people to enable audit logs to debug this SNI stuff?
+		// Using Info level so the user can safely configure a production Supervisor to show this message if they choose.
+		plog.Info("received incoming request",
+			"proto", req.Proto,
+			"method", req.Method,
+			"host", req.Host,
+			"requestSNIServerName", requestutil.SNIServerName(req),
+			"path", req.URL.Path,
+			"remoteAddr", req.RemoteAddr,
+			"userAgent", req.UserAgent(),
+			"foundFederationDomainRequestHandler", requestHandler != nil,
+		)
```

Comment:
We should resolve or remove this todo before merging this branch. If audit logging is always enabled, then perhaps we don't need this old log statement anymore, as long as the audit logs contain the same information about the incoming requests.

Reply:
Done! Because the audit logs will always show requests that 404, and otherwise have the same key/values as this old message, then we don't need this old message anymore.